spec/strava/api/client/endpoints/uploads/create_upload_spec.rb

```diff
@@ -5,18 +5,66 @@
 RSpec.describe 'Strava::Api::Client#upload_activity' do
   include_context 'API client'
   let(:file) { 'spec/fixtures/strava/files/17611540601.tcx' }
-  it 'uploads an activity', vcr: { cassette_name: 'client/create_upload' } do
+  it 'uploads an activity successfully', vcr: { cassette_name: 'client/create_upload' } do
     upload = client.create_upload(
       file: Faraday::UploadIO.new(file, 'application/tcx+xml'),
       name: 'Uploaded Activity',
       description: 'Uploaded by strava-ruby-client.',
-      data_type: 'tcx',
-      external_id: 'strava-ruby-client-upload-1'
+      data_type: 'tcx'
     )
-    expect(upload.id).to eq 2_136_460_097
-    expect(upload.external_id).to eq 'strava-ruby-client-upload-1.tcx'
-    expect(upload.error).to eq nil
-    expect(upload.status).to eq 'Your activity is still being processed.'
-    expect(upload.activity_id).to eq nil
+
+    expect(upload.id).to eq(8_914_982_151)
+    expect(upload.error).to be(nil)
+    expect(upload.status).to eq('Your activity is still being processed.')
+    expect(upload.activity_id).to eq(nil)
+
+    # strava needs some time, to process the uploaded file, loop until processing finished
+    upload_status = client.upload(upload.id)
+    while upload_status.processing?
+      sleep(1)
+      upload_status = client.upload(upload.id)
+    end
+
+    expect(upload_status).to be_a(Strava::Models::Upload)
+    expect(upload_status.id).to be(8_914_982_151)
+    expect(upload_status.processed?).to be(true)
+    expect(upload_status.processing?).to be(false)
+    expect(upload_status.external_id).to eql('17611540601.tcx')
+    expect(upload_status.activity_id).to be(8_315_061_462)
+  end
+
+  it 'upload processing', vcr: { cassette_name: 'client/create_upload_file_processing' } do
+    # You need to wait when recording new cassettes, for the background job processing the upload to finish.
```

Comment:
You can use a wait here as well, no? I would remove these comments since the `sleep` is pretty self-explanatory.

Reply:
no, in this case to log the "in process" state one needs to query right away. `processing?` will evaluate to `true`, meaning neither success nor failure.

I updated spec descriptions and fixed comments, also added a sentence in README.md to clarify things a little better.